lib/devices/phone_device_manager.js

```diff
@@ -0,0 +1,78 @@
+import assert from 'assert';
+import Debug from 'debug';
+
+import Testdroid from 'testdroid-client';
+
+let debug = Debug('taskcluster-docker-worker:devices:phoneManager');
+
+export default class PhoneDeviceManager {
+  constructor(config) {
+    assert(config.deviceManagement.phone.sims, "Must supply the number of sims required");
+    assert(config.deviceManagement.phone.type, "Must supply the type of phone");
+    assert(config.testdroid, "Must supply testdroid configuration");
+    assert(config.testdroid.url, "Must supply testdroid cloud url");
+    assert(config.testdroid.username, "Must supply testdroid cloud username");
+    assert(config.testdroid.password, "Must supply testdroid cloud password");
+
+    this.config = config;
+    this.client = new Testdroid(
+        config.testdroid.url,
+        config.testdroid.username,
+        config.testdroid.password
+    );
+    this.deviceFilter = {
+      'type': config.deviceManagement.phone.type,
+      'sims': config.deviceManagement.phone.sims
+    };
+  }
+
+  async devices() {
+    let devices = await this.client.getDevices(this.deviceFilter);
```

Comment:
I suspect it's best not to poll this every 5s, so maybe implement `maxUpdateInterval` as suggested above...

And just return the previous list without modifications if `maxUpdateInterval` hasn't passed... Ie. it's probably a time that you have to count using Date.now()...

Note, if the list of devices contains available devices we have a problem...
 1. we can't really update every 5s to see if it's still available.
 2. we can't really assume it's still available.

For now, this might not be a problem... because each worker can handle ~10 devices (if I recall you correctly), so we're not going to have many works, hence, polling every 5s might work.